design-documents/asynchronous-import/modularity/data-exchanging.md

```diff
@@ -0,0 +1,211 @@
+# Data exchanging with Magento instance
+
+## Modules
+
+- `AsynchronousImportDataExchangingApi`
+- `AsynchronousImportDataExchanging`
+
+## Implementation
+
+### API
+
+```php
+namespace Magento\AsynchronousImportDataExchangingApi\Api;
+
+use Magento\AsynchronousImportDataExchangingApi\Api\Data\ImportInterface;
+use Magento\Framework\Validation\ValidationException;
+
+/**
+ * Operation for exchanging import data with destination instance. Uses differect strategies for data import
+ *
+ * @api
+ */
+interface ExchangeImportDataInterface
+{
+    /**
+     * Operation for exchanging import data with destination instance
+     *
+     * @param ImportInterface $import
+     * @param array $importData
+     * @return void
+     * @throws ValidationException
+     * @throws ImportDataExchangeException
+     */
+    public function execute(ImportInterface $import, array $importData): void;
+}
+```
+
+```php
+namespace Magento\AsynchronousImportDataExchangingApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+/**
+ * Describes how to import data
+ *
+ * @api
+ */
+interface ImportInterface extends ExtensibleDataInterface
+{
+    /**
+     * Get import uuid
+     *
+     * @return string|null
+     */
+    public function getUuid(): ?string;
```

Comment:
do we need to generalize UUID? can't we just say ID. The problem with UUID is that they are big in size and doesn't index well. When your data grows in the table, the index size also grows with it, which leads to query performance hit.

https://www.callicoder.com/distributed-unique-id-sequence-number-generator/

Reply:
I do not expect really big data flow where, even if you are doing daily imports with 100+ files daily (which is not common case), we would need to have some cleaning functionality for such table. 
to use UUID was provided in Asynchronous operations and as I know idea of Arch team was to go into UUID direction and discussion is still open, or?

Reply:
there some other PRs currently opened to replace DB Sequence for IDs with some other strategy, when the ID is not externally provided; where we are having some discussions!